### PR TITLE
[wasm][debugger] Adding support to receive evaluationOptions from dap.

### LINKED ISSUE #49144: [wasm][debugger] Properties in the watch window show "𝑓 <Property>" instead of the value of the property
### Describe the bug
Watch window does not show the value of properties for Blazor WASM projects, requiring you instead to navigate to and expand each property to see its value.

### To Reproduce
1. Create a new Blazor WASM project
2. Add a new .NET library project to the solution
3. Add the `Foo` class to the library with the code below
4. Add a reference to the .NET library project from the WASM project
5. Update the `FetchData.razor` page with the code below.
6. Add a breakpoint in the `Bar` property
7. Start debugging
8. Click on the "Fetch Data" link in the window
9. The program should break in the `Foo` class. Open the locals Window

### Expected Results
`this.Bar` will show the value `sample-data/weather.json`
image
(Image shown is the locals window when the same value is viewed from a console project)

### Actual Results
`this.Bar` shows `𝑓 Bar() {}` and requires you to expand the property to view the value
image

#### Foo.cs
```csharp
using System;

namespace BlazorBugsLib
{
  public class Foo
  {
    public string Bar => "sample-data/weather.json";
    public string Lorem { get; set; } = "Safe";
    public string Ipsum { get; set; } = "Side";
    public Something What { get; } = new Something();
  }

  public class Something
  {
    public string Name { get; set; }
    public Something() => Name = "Name of something";
    public override string ToString() => Name;
  }
}
```

#### FetchData.razor
```csharp
  protected override async Task OnInitializedAsync()
  {
    forecasts = await Http.GetFromJsonAsync<WeatherForecast[]>(new BlazorBugsLib.Foo().Bar);
  }
```

### PR DESCRIPTION
**Pull request by thaystg**
To support receiving EvaluationOptions from dap.

Then we will be able to auto evaluate properties or not based on what is defined on VS settings.
https://user-images.githubusercontent.com/4503299/176544439-3ae898e5-4eb6-4883-bd57-a277fb19ff30.mp4

Depends on:
https://github.com/microsoft/vscode-js-debug/pull/1323

With this check disabled: 
image
image

With this check enabled:
image
image

 Fixes https://github.com/dotnet/runtime/issues/49144

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/MemberObjectsExplorer.cs [hunk 1]
```diff
@@ -558,7 +558,7 @@ JObject GetNotAutoExpandableObject(int methodId, string propertyName)
                     typeId,
                     typeName,
                     getPropertiesParamBuffer,
-                    getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerProxyAttribute),
+                    getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerProxyAttribute) || getCommandType.HasFlag(GetObjectCommandOptions.AutoEvaluateProperties),
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/MemberObjectsExplorer.cs:561]:
```suggestion
                    isAutoExpandable: getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerProxyAttribute) || getCommandType.HasFlag(GetObjectCommandOptions.AutoEvaluateProperties),
```

**Comment by radical** [comment]
Can we add a test where it sends the setting, and we check if properties are being auto expanded or not?

**Comment by thaystg** [comment]
For sure we can do it, I just didn't add it yet, because I'm waiting if Connor will approve the changes the way it is implemented on the PR on vscode-js-debug repo.

**Comment by thaystg** [comment]
As I discussed with Larry we would like to merge it on our side and hope to get the merge on VS side without any change.
I'll add the tests in the PR and we can merge both of them before RC1.

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
should we land this in main now?

**Comment by marek-safar** [comment]
@thaystg what's the plan with this PR?

**Comment by thaystg** [comment]
Still waiting for the approval of the PR on vscode-js-debug repo: https://github.com/microsoft/vscode-js-debug/pull/1323

**Review by ilonatommy — APPROVED** [review]

**Pull request merged**